controllers/drcluster_controller_test.go

```diff
@@ -186,6 +186,42 @@ func inspectClusterManifestSubscriptionCSV(match bool, value string, drcluster *
 	}
 }
 
+func updateDRClusterParameters(clusterName string, clusterFence ramen.ClusterFenceState,
+	cidrs []string, s3ProfileName string,
+) *ramen.DRCluster {
+	drclusterLookupKey := types.NamespacedName{
+		Name: clusterName,
+	}
+	latestdrc := &ramen.DRCluster{}
+
+	retryErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		var err error
+
+		err = apiReader.Get(context.TODO(), drclusterLookupKey, latestdrc)
+		if err != nil {
+			return err
+		}
+
+		latestdrc.Spec.ClusterFence = clusterFence
+
+		if s3ProfileName != "" {
+			latestdrc.Spec.S3ProfileName = s3ProfileName
+		}
+
+		if cidrs != nil {
+			latestdrc.Spec.CIDRs = cidrs
+		}
+
+		err = k8sClient.Update(context.TODO(), latestdrc)
+
+		return err
+	})
+
+	Expect(retryErr).NotTo(HaveOccurred())
+
+	return latestdrc
```

Comment:
...but this may be zero DRCluster if all retries failed. What happens when Expect() fails because retryErr is not nil? Does expect panic and the panic is handled by
the test framework?

It may be more clear if updateDRClusterParameters() return DRCluster, error, and the expect code is done in side the test.

Reply:
This test will fail with error and exit